lib/config.go

```diff
@@ -68,7 +69,8 @@ func flagsToConfig(f *flag.FlagSet) (*Config, error) {
 	f.StringVar(&cfg.CDNDistributionID, "distribution-id", "", "optional CDN distribution ID for cache invalidation")
 	f.StringVar(&cfg.ConfigFile, "config", ".s3deploy.yml", "optional config file")
 	f.IntVar(&cfg.MaxDelete, "max-delete", 256, "maximum number of files to delete per deploy")
-	f.BoolVar(&cfg.PublicReadACL, "public-access", false, "set public ACL on uploaded objects, defaults to private if not set.")
+	f.BoolVar(&cfg.PublicReadACL, "public-access", false, "DEPRECATED: please set -acl='public-read'")
```

Comment:
This PR marks the `public-access` flag as deprecated. The reason I did this is because the `acl` flag conflicts with this shorthand. For example, this is an invalid state:

```
 ./s3deploy -bucket=myBucket -acl=bucket-owner-full-control -public-access=true
s3deploy v2, commit none, built at unknown
error: you passed a value for the flags public-access and acl, which is not supported. the public-access flag is deprecated. please use the acl flag moving forward
```

The `acl` flag is a bit less user-friendly, so I could see an argument for trying to keep the `-public-access` flag in the longer-term.

Either way, I wanted to make sure existing users could opt-in to the new behavior and have time to switch over to the new flag before it's removed.